src/mo_frontend/typing.ml

```diff
@@ -3042,7 +3082,7 @@ and is_typ_dec dec : bool = match dec.it with
 
 and infer_obj env obj_sort exp_opt dec_fields at : T.typ =
   let s = obj_sort.it in
```

Comment:
Did you consider may exanding out the dec_field to include all Included dec_fields before doing this check?